### PR TITLE
fix: make sure to do the dom mutation inside a `deferMutate`

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

**Comment by ampsauce** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 1fef51ba1455c7c3bdc1a2f249358f60f00089c6
Build details: https://travis-ci.org/ampsauce/amphtml/builds/89319706

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

File: builtins/amp-ad.js [hunk 1]
```diff
@@ -253,13 +253,17 @@ export function installAd(win) {
...
 
     noContentHandler_() {
-      if (this.isDefaultFallback_) {
-        this.setDefaultFallback_();
-        this.element.appendChild(this.fallback_);
-      }
-
-      this.element.removeChild(this.iframe_);
-      this.toggleFallback(true);
+      // TODO: (erwinm) check the performance of the append here
```
Comment (dvoytenko) [inline builtins/amp-ad.js]: That's basically it. TODO is done :)

**Comment by dvoytenko** [comment]
LGTM

File: builtins/amp-ad.js [hunk 2]
```diff
@@ -245,7 +245,7 @@ export function installAd(win) {
 
         // Triggered by context.noContentAvailable() inside the ad iframe.
         const unlisten = listen(this.iframe_, 'no-content', () => {
-          this.noContentHandler_();
+          this.deferMutate(this.noContentHandler_.bind(this));
```
Comment (erwinmombay) [inline builtins/amp-ad.js]: @dvoytenko moved the deferMutate call here instead to make my test pass.

**Comment by ampsauce** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 75eb29070795485855bf750b29be2469730833c1
Build details: https://travis-ci.org/ampsauce/amphtml/builds/89323625

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Pull request merged**